### PR TITLE
Add Ray User Guide

### PR DESCRIPTION
**Pull request by hkvision**

### CODE REVIEW

**Comment by hkvision** [comment]
https://hkvision-ray.readthedocs.io/en/latest/doc/UserGuide/ray.html Take a look @jason-dai

File: readthedocs/source/doc/UserGuide/ray.md [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+_**Note:** Analytics Zoo has been tested on Ray 0.8.4 and you are highly recommended to use this tested version._
+
+
+### **1. Install**
+
+Follow the guide [here](./python.html#install) for environment preparation. When installing analytics-zoo with pip, you can specify the extras key `[ray]` to additionally install the additional dependencies essential for running Ray (i.e. `ray==0.8.4`, `psutil`, `aiohttp`, `setproctitle`, `pyarrow==0.17.0`):
+
+```bash
+pip install analytics-zoo[ray]
+```
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/ray.md:16]: include `coda` in the commands

File: readthedocs/source/doc/UserGuide/ray.md [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+- You can retrieve the information of the Ray cluster via [`OrcaContext`](../Orca/Overview/orca-context.md):
+
+  ```python
+  from zoo.orca import OrcaContext
+  
+  ray_ctx = OrcaContext.get_ray_context()
+  address_info = ray_ctx.address_info  # The dictionary information of the ray cluster, including node_ip_address, object_store_address, webui_url, etc.
+  redis_address = ray_ctx.redis_address  # The redis address of the ray cluster.
+  ```
+
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/ray.md:61]: add stop_orca_context

File: readthedocs/source/doc/UserGuide/ray.md [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+Follow the guide [here](./python.html#install) for environment preparation. When installing analytics-zoo with pip, you can specify the extras key `[ray]` to additionally install the additional dependencies essential for running Ray (i.e. `ray==0.8.4`, `psutil`, `aiohttp`, `setproctitle`, `pyarrow==0.17.0`):
+
+```bash
+pip install analytics-zoo[ray]
+```
+
+---
+### **2. Initialize**
+
+We recommend using `init_orca_context` to initiate and run Analytics Zoo on the underlying cluster. The Ray cluster would be launched as well by specifying `init_ray_on_spark=True`. For example, to launch Spark and Ray on standard Hadoop/YARN clusters in [YARN client mode](https://spark.apache.org/docs/latest/running-on-yarn.html#launching-spark-on-yarn):
```
Comment (jason-dai) [inline readthedocs/source/doc/UserGuide/ray.md:21]: add reference to Orca Context

Comment (hkvision) [reply readthedocs/source/doc/UserGuide/ray.md:27]: Added.

File: readthedocs/source/doc/UserGuide/ray.md (hunk 1 above)
Comment (hkvision) [reply readthedocs/source/doc/UserGuide/ray.md:20]: Added. Take a look.

File: readthedocs/source/doc/UserGuide/ray.md (hunk 2 above)
Comment (hkvision) [reply readthedocs/source/doc/UserGuide/ray.md:68]: Added.

**Comment by hkvision** [comment]
The updated link: https://hkvision-ray.readthedocs.io/en/latest/doc/UserGuide/ray.html @jason-dai

**Comment by jason-dai** [comment]
LGTM

**Comment by hkvision** [comment]
Do I still need to update https://github.com/jason-dai/analytics-zoo#getting-started-with-rayonspark? or you just link the user guide to this readthedocs page? @jason-dai

**Comment by jason-dai** [comment]
> Do I still need to update https://github.com/jason-dai/analytics-zoo#getting-started-with-rayonspark? or you just link the user guide to this readthedocs page? @jason-dai

Please submit a PR with a short example.

**Comment by hkvision** [comment]
> > Do I still need to update https://github.com/jason-dai/analytics-zoo#getting-started-with-rayonspark? or you just link the user guide to this readthedocs page? @jason-dai
> 
> Please submit a PR with a short example.

okay

**Comment by hkvision** [comment]
https://github.com/jason-dai/analytics-zoo/pull/3 @jason-dai

**Pull request merged**